In [2]:
import numpy as np
import pandas as pd

In [18]:
data=pd.read_excel('marketting_Campaign.xlsx')#reading excelfile

In [198]:
dt=pd.DataFrame(data)#converting it into a dtaframe
df=dt.to_csv('data.csv')#then it converts to csv file
df=pd.read_csv('data.csv')#reading csv file
df

Unnamed: 0  Column1  Year_Birth   Education Marital_Status   Income  \
0              0     5524        1957  Graduation         Single  58138.0   
1              1     2174        1954  Graduation         Single  46344.0   
2              2     4141        1965  Graduation       Together  71613.0   
3              3     6182        1984  Graduation       Together  26646.0   
4              4     5324        1981         PhD        Married  58293.0   
...          ...      ...         ...         ...            ...      ...   
2235        2235    10870        1967  Graduation        Married  61223.0   
2236        2236     4001        1946         PhD       Together  64014.0   
2237        2237     7270        1981  Graduation       Divorced  56981.0   
2238        2238     8235        1956      Master       Together  69245.0   
2239        2239     9405        1954         PhD        Married  52869.0   

      Kidhome  Teenhome Dt_Customer  Recency  ...  NumWebVisitsMonth  \
0           0         0  2012-09-04       58  ...                  7   
1           1         1  2014-03-08       38  ...                  5   
2           0         0  2013-08-21       26  ...                  4   
3           1         0  2014-02-10       26  ...                  6   
4           1         0  2014-01-19       94  ...                  5   
...       ...       ...         ...      ...  ...                ...   
2235        0         1  2013-06-13       46  ...                  5   
2236        2         1  2014-06-10       56  ...                  7   
2237        0         0  2014-01-25       91  ...                  6   
2238        0         1  2014-01-24        8  ...                  3   
2239        1         1  2012-10-15       40  ...                  7   

      AcceptedCmp3  AcceptedCmp4  AcceptedCmp5  AcceptedCmp1  AcceptedCmp2  \
0                0             0             0             0             0   
1                0             0             0             0             0   
2                0             0             0             0             0   
3                0             0             0             0             0   
4                0             0             0             0             0   
...            ...           ...           ...           ...           ...   
2235             0             0             0             0             0   
2236             0             0             0             1             0   
2237             0             1             0             0             0   
2238             0             0             0             0             0   
2239             0             0             0             0             0   

      Complain  Z_CostContact  Z_Revenue  Response  
0            0              3         11         1  
1            0              3         11         0  
2            0              3         11         0  
3            0              3         11         0  
4            0              3         11         0  
...        ...            ...        ...       ...  
2235         0              3         11         0  
2236         0              3         11         0  
2237         0              3         11         0  
2238         0              3         11         0  
2239         0              3         11         1  

[2240 rows x 30 columns]

In [200]:
print(df.isnull().sum())#printing sum of null value cells

Unnamed: 0              0
Column1                 0
Year_Birth              0
Education               0
Marital_Status          0
Income                 24
Kidhome                 0
Teenhome                0
Dt_Customer             0
Recency                 0
MntWines                0
MntFruits               0
MntMeatProducts         0
MntFishProducts         0
MntSweetProducts        0
MntGoldProds            0
NumDealsPurchases       0
NumWebPurchases         0
NumCatalogPurchases     0
NumStorePurchases       0
NumWebVisitsMonth       0
AcceptedCmp3            0
AcceptedCmp4            0
AcceptedCmp5            0
AcceptedCmp1            0
AcceptedCmp2            0
Complain                0
Z_CostContact           0
Z_Revenue               0
Response                0
dtype: int64


In [79]:
df=df.fillna("unknown")#fill the null value cells with unknown
print(df.isnull().sum())#printing to check null value exists

Unnamed: 0             0
Column1                0
Year_Birth             0
Education              0
Marital_Status         0
Income                 0
Kidhome                0
Teenhome               0
Dt_Customer            0
Recency                0
MntWines               0
MntFruits              0
MntMeatProducts        0
MntFishProducts        0
MntSweetProducts       0
MntGoldProds           0
NumDealsPurchases      0
NumWebPurchases        0
NumCatalogPurchases    0
NumStorePurchases      0
NumWebVisitsMonth      0
AcceptedCmp3           0
AcceptedCmp4           0
AcceptedCmp5           0
AcceptedCmp1           0
AcceptedCmp2           0
Complain               0
Z_CostContact          0
Z_Revenue              0
Response               0
dtype: int64


In [81]:
df=df.drop_duplicates()#drop duplicates

In [124]:
df['Marital_Status']=df['Marital_Status'].str.title().str.strip()
df['Education']=df['Education'].str.title().str.strip()

In [202]:
#converting Dt_Customer to date data type
df['Dt_Customer']= pd.to_datetime(df['Dt_Customer'])
print(df['Dt_Customer'])

0      2012-09-04
1      2014-03-08
2      2013-08-21
3      2014-02-10
4      2014-01-19
          ...    
2235   2013-06-13
2236   2014-06-10
2237   2014-01-25
2238   2014-01-24
2239   2012-10-15
Name: Dt_Customer, Length: 2240, dtype: datetime64[ns]


In [210]:
#creating a new column names customer since days it calculates how long they have customer
today=pd.to_datetime('today')
df['Customer_since_years']=(today-df['Dt_Customer']).dt.days//365
df.head()

Unnamed: 0  Column1  Year_Birth   Education Marital_Status   Income  \
0           0     5524        1957  Graduation         Single  58138.0   
1           1     2174        1954  Graduation         Single  46344.0   
2           2     4141        1965  Graduation       Together  71613.0   
3           3     6182        1984  Graduation       Together  26646.0   
4           4     5324        1981         PhD        Married  58293.0   

   Kidhome  Teenhome Dt_Customer  Recency  ...  AcceptedCmp4  AcceptedCmp5  \
0        0         0  2012-09-04       58  ...             0             0   
1        1         1  2014-03-08       38  ...             0             0   
2        0         0  2013-08-21       26  ...             0             0   
3        1         0  2014-02-10       26  ...             0             0   
4        1         0  2014-01-19       94  ...             0             0   

   AcceptedCmp1  AcceptedCmp2  Complain  Z_CostContact  Z_Revenue  Response  \
0             0             0         0              3         11         1   
1             0             0         0              3         11         0   
2             0             0         0              3         11         0   
3             0             0         0              3         11         0   
4             0             0         0              3         11         0   

   Customer_since_years  Age  
0                    12   68  
1                    11   71  
2                    11   60  
3                    11   41  
4                    11   44  

[5 rows x 32 columns]

In [208]:
#creating a new column named Age which calculates age of the customer
df['Age']=today.year-df['Year_Birth']
df.head()

Unnamed: 0  Column1  Year_Birth   Education Marital_Status   Income  \
0           0     5524        1957  Graduation         Single  58138.0   
1           1     2174        1954  Graduation         Single  46344.0   
2           2     4141        1965  Graduation       Together  71613.0   
3           3     6182        1984  Graduation       Together  26646.0   
4           4     5324        1981         PhD        Married  58293.0   

   Kidhome  Teenhome Dt_Customer  Recency  ...  AcceptedCmp4  AcceptedCmp5  \
0        0         0  2012-09-04       58  ...             0             0   
1        1         1  2014-03-08       38  ...             0             0   
2        0         0  2013-08-21       26  ...             0             0   
3        1         0  2014-02-10       26  ...             0             0   
4        1         0  2014-01-19       94  ...             0             0   

   AcceptedCmp1  AcceptedCmp2  Complain  Z_CostContact  Z_Revenue  Response  \
0             0             0         0              3         11         1   
1             0             0         0              3         11         0   
2             0             0         0              3         11         0   
3             0             0         0              3         11         0   
4             0             0         0              3         11         0   

   Customer_since_years  Age  
0                    12   68  
1                    11   71  
2                    11   60  
3                    11   41  
4                    11   44  

[5 rows x 32 columns]

In [212]:
#replacing columns space with underscores and change to lowercase
df.columns=df.columns.str.lower().str.replace(' ','_')
df.head()

unnamed:_0  column1  year_birth   education marital_status   income  \
0           0     5524        1957  Graduation         Single  58138.0   
1           1     2174        1954  Graduation         Single  46344.0   
2           2     4141        1965  Graduation       Together  71613.0   
3           3     6182        1984  Graduation       Together  26646.0   
4           4     5324        1981         PhD        Married  58293.0   

   kidhome  teenhome dt_customer  recency  ...  acceptedcmp4  acceptedcmp5  \
0        0         0  2012-09-04       58  ...             0             0   
1        1         1  2014-03-08       38  ...             0             0   
2        0         0  2013-08-21       26  ...             0             0   
3        1         0  2014-02-10       26  ...             0             0   
4        1         0  2014-01-19       94  ...             0             0   

   acceptedcmp1  acceptedcmp2  complain  z_costcontact  z_revenue  response  \
0             0             0         0              3         11         1   
1             0             0         0              3         11         0   
2             0             0         0              3         11         0   
3             0             0         0              3         11         0   
4             0             0         0              3         11         0   

   customer_since_years  age  
0                    12   68  
1                    11   71  
2                    11   60  
3                    11   41  
4                    11   44  

[5 rows x 32 columns]

In [178]:
#removing duplicated columns
df = df.loc[:, ~df.columns.duplicated()]

In [214]:
#reodering some columns for readability 
main_cols = ['year_birth', 'age', 'education', 'marital_status', 'income', 
        'kidhome', 'teenhome', 'dt_customer', 'customer_since_years'] 

remaining_cols=[col for col in df.columns if col not in main_cols]
                
df = df[main_cols+remaining_cols]

df.head()

year_birth  age   education marital_status   income  kidhome  teenhome  \
0        1957   68  Graduation         Single  58138.0        0         0   
1        1954   71  Graduation         Single  46344.0        1         1   
2        1965   60  Graduation       Together  71613.0        0         0   
3        1984   41  Graduation       Together  26646.0        1         0   
4        1981   44         PhD        Married  58293.0        1         0   

  dt_customer  customer_since_years  unnamed:_0  ...  numwebvisitsmonth  \
0  2012-09-04                    12           0  ...                  7   
1  2014-03-08                    11           1  ...                  5   
2  2013-08-21                    11           2  ...                  4   
3  2014-02-10                    11           3  ...                  6   
4  2014-01-19                    11           4  ...                  5   

   acceptedcmp3  acceptedcmp4  acceptedcmp5  acceptedcmp1  acceptedcmp2  \
0             0             0             0             0             0   
1             0             0             0             0             0   
2             0             0             0             0             0   
3             0             0             0             0             0   
4             0             0             0             0             0   

   complain  z_costcontact  z_revenue  response  
0         0              3         11         1  
1         0              3         11         0  
2         0              3         11         0  
3         0              3         11         0  
4         0              3         11         0  

[5 rows x 32 columns]

In [186]:
df.to_csv("cleaned_data.csv",index=False)

# Marketing Campaign Data Cleaning Project
---

## **Dataset Overview**
The raw dataset contains **2240 rows x 30 columns** with features like:
- Customer demographics (`Year_Birth`, `Education`, `Marital_Status`, `Income`)
- Purchase behavior (`MntWines`, `MntFruits`, `NumWebPurchases`)
- Campaign responses (`AcceptedCmp1-5`, `Response`)
- Temporal data (`Dt_Customer`, `Recency`)

**Cleaned Dataset Output**:  
`cleaned_data.csv` (2240 rows x 32 columns) includes new features:
- `age`: Customer age derived from `Year_Birth`.
- yearstomer_since_days`: Years since the  actually years).
- Standardized column names (lowercase wiepository Structure**